lib/db/set.go

```diff
@@ -470,10 +476,19 @@ func DropDeltaIndexIDs(db *Lowlevel) {
 	}
 }
 
-func normalizeFilenames(fs []protocol.FileInfo) {
-	for i := range fs {
-		fs[i].Name = osutil.NormalizedFilename(fs[i].Name)
+func normalizeFilenamesAndDropDuplicates(fs []protocol.FileInfo) []protocol.FileInfo {
+	positions := make(map[string]int, len(fs))
+	newFs := fs[:0]
+	for _, f := range fs {
+		norm := osutil.NormalizedFilename(f.Name)
+		if i, ok := positions[norm]; ok {
+			newFs = append(newFs[:i], newFs[i+1:]...)
```

Comment:
Problem: this changes the positions of everything after i, so the map is now wrong for those items. Maybe replace the item to be dropped with a sentinel value, and then filter these in a latter stage?

Reply:
(And try to make a couple of nasty test cases for just this function?)